code/game/machinery/PDApainter.dm

```diff
@@ -30,15 +30,11 @@
 /obj/machinery/pdapainter/New()
 	..()
 	var/blocked = list(/obj/item/pda/silicon/ai, /obj/item/pda/silicon/robot, /obj/item/pda/silicon/pai, /obj/item/pda/heads,
-						/obj/item/pda/clear, /obj/item/pda/syndicate)
+						/obj/item/pda/clear, /obj/item/pda/syndicate, /obj/item/pda/chameleon, /obj/item/pda/chameleon/broken)
 
-	for(var/P in typesof(/obj/item/pda)-blocked)
-		var/obj/item/pda/D = new P
-
-		//D.name = "PDA Style [colorlist.len+1]" //Gotta set the name, otherwise it all comes up as "PDA"
-		D.name = D.icon_state //PDAs don't have unique names, but using the sprite names works.
-
-		src.colorlist += D
+	for(var/thing in typesof(/obj/item/pda) - blocked)
+		var/obj/item/pda/P = thing
+		colorlist[initial(P.icon_state)] = initial(P.desc)
```

Comment:
This feels like it should be the other way round, or have P.icon_state as the key and value.

Reply:
This definitely appears as `pda_clown`, `pda_toxins`, and what have you, which is the same behavior to the player.

Reply:
Ok thats fine